thrust/system/cuda/detail/bulk/malloc.hpp

```diff
@@ -38,9 +38,24 @@ inline __device__ bool is_on_chip(void *ptr)
 template<typename T>
 inline __device__ T *on_chip_cast(T *ptr)
 {
+#ifndef __clang__
```

Comment:
Please make it #if !(defined (__clang__) && defined(__CUDA__)). If clang is  a host compiler, the wrong code path will be taken with  nvcc.